app/src/main/java/com/iic/shopingo/ui/ContactDetailsActivity.java

```diff
@@ -26,17 +30,43 @@
   @InjectView(R.id.contact_phone_editview)
   EditText phoneEditView;
 
+  @InjectView(R.id.contact_avatar_imageview)
+  ImageView avatarImageView;
+
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_contact_details);
     ButterKnife.inject(this);
+
+    prefillFields();
   }
 
   @OnClick(R.id.contact_save_btn)
   void onSaveClicked() {
+    updateUser();
     Intent intent = new Intent(this, HomeActivity.class);
     intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
     startActivity(intent);
   }
+
+  private void updateUser() {
+    User user = SharedUserConnector.getInstance().getCurrentUser();
+    user.setFirstName(firstNameEditView.getText().toString());
+    user.setLastName(lastNameEditView.getText().toString());
+    user.setStreet(streetEditView.getText().toString());
+    user.setCity(cityEditView.getText().toString());
+    user.setPhoneNumber(phoneEditView.getText().toString());
+    SharedUserConnector.getInstance().setCurrentUser(user);
```

Comment:
Add a `TODO` for input validation?
All fields are required and the phone number should be valid.